libgloss/arc/hl/hl_close.c

```diff
@@ -0,0 +1,41 @@
+/*********************************************************************
+Copyright  2002-2020 Synopsys Inc.
+*********************************************************************/
+
+#include <errno.h>
+#include <stdint.h>
+#include <unistd.h>
+
+#include "hl_toolchain.h"
+#include "hl_api.h"
+
+static int
+hl_close (int fd)
+{
+  int32_t ret;
+  uint32_t host_errno;
+  volatile __uncached char *p;
+
+  p = hl_message (HL_SYSCALL_CLOSE, "i:ii",
+                  (uint32_t) fd,           /* i */
+                  (uint32_t *) &ret,       /* :i */
+                  (uint32_t *) &host_errno /* :i */);
+
+  if (p == NULL) {
+    errno = ETIMEDOUT;
+    ret = -1;
+  } else if (ret < 0) {
+    errno = host_errno;
+    ret = -1;
+  }
+
+  hl_delete ();
+
+  return ret;
+}
+
+int
+_close (int fd)
```

Comment:
like in the case of argv

Reply:
see above

Reply:
inline if you don't like to change it. The best to see what I mean, is to compile a file with different compiler and/or cpu options and wander how you can bridge to a minimal and yet "beautiful" implementation.

Reply:
done